argmin/src/core/state/iterstate.rs

```diff
@@ -96,7 +98,7 @@ where
     ///
     /// ```
     /// # use argmin::core::{IterState, State};
-    /// # let state: IterState<Vec<f64>, (), (), (), f64> = IterState::new();
+    /// # let state: IterState<Vec<f64>, (), (), (), f64, Vec<f64>> = IterState::new();
```

Comment:
When the residuals are not part of the doc test, please set `R = ()`. (also applies to the docs below)